arctouch-challenge-ios/ViewModels/MovieDetailViewModel.swift

```diff
@@ -0,0 +1,40 @@
+//
+//  MovieDetailViewModel.swift
+//  arctouch-challenge-ios
+//
+//  Created by Pedro H J Fabrino on 11/03/18.
+//  Copyright © 2018 AIS Digital. All rights reserved.
+//
+
+import UIKit
+import RxSwift
+import RxCocoa
+
+class MovieDetailViewModel: BaseViewModel {
+    
+    var movie: BehaviorRelay<Movie>
+    var service: MovieService
+    
+    init(movieService: MovieService = MovieService()) {
+        self.service = movieService
+        self.movie = BehaviorRelay<Movie>(value: Movie())
+        super.init()
+    }
+    
+    func loadDetails(for movie: Movie) {
+        guard let id = movie.id else {
+            return
+        }
+        _ = service.details(for: id).subscribe({ (event) in
+            switch (event) {
+            case .next(let detailedMovie):
+                self.movie.accept(detailedMovie)
+            
+            case .error(let error):
+                Log.error("Error: \(error.localizedDescription)")
```

Comment:
I would create another behaviourRelay, an BehaviourRelay<Error?>, and would subscribe to it on any needed view. Whenever I got an error in that log, go through error.accept(error) and let the view show to the user the way it can. In that case, I would show an alert to the user, so he can try again in a few minutes.